ivc/src/ivc/interpreter.rs

```diff
@@ -286,20 +287,22 @@ where
     // Left-Right-Output for a given limb size.
     fn repack_output<F: PrimeField, const TWO_LIMB_SIZE: usize, const N_COL_TOTAL: usize>(
         input: [Vec<Vec<F>>; 3],
-    ) -> [[[F; TWO_LIMB_SIZE]; N_COL_TOTAL]; 3] {
-        input
-            .into_iter()
-            .map(|vector: Vec<Vec<_>>| {
-                vector
-                    .into_iter()
-                    .map(|subvec: Vec<_>| subvec.try_into().unwrap())
-                    .collect::<Vec<_>>()
-                    .try_into()
-                    .unwrap()
-            })
-            .collect::<Vec<_>>()
-            .try_into()
-            .unwrap()
+    ) -> Box<[[[F; TWO_LIMB_SIZE]; N_COL_TOTAL]; 3]> {
```

Comment:
Wondering if we could get rid of `Vec<Vec<F>>` and always use fixed size array and/or flat array. This code adds noise for the reader.

Reply:
Yeah, in the following PR I introduce some converting methods to use multi-dimensional arrays more. Might be good to use more commonly indeed :)

Reply:
Opened https://github.com/o1-labs/proof-systems/issues/2253, just to keep track of it, not a priority.